src/select.test.ts

```diff
@@ -0,0 +1,222 @@
+  });
+
+  it("creates query from {condition: q()} composition", () => {
+    const { query, schema } = q.__experimental_select({
+      "foo > 2": q("bar").grab({
+        bar: z.boolean(),
+      }),
+      default: {
+        baz: z.string(),
+      },
+    });
+
+    expect(query).toBe("select(foo > 2 => bar{bar}, { baz })");
+    expectTypeOf({} as z.infer<typeof schema>).toEqualTypeOf<
+      { bar: boolean } | { baz: string }
+    >();
+  });
+
+  it("creates query from {condition: [name, schema]} composition", () => {
+    const { query, schema } = q.__experimental_select({
+      "foo > 2": ["bar", q.boolean()],
+      default: {
+        baz: z.string(),
+      },
+    });
+
+    expect(query).toBe("select(foo > 2 => bar, { baz })");
+    expectTypeOf({} as z.infer<typeof schema>).toEqualTypeOf<
+      boolean | { baz: string }
+    >();
+  });
+
+  it("makes schema nullable when default condition omitted", () => {
+    const { query, schema } = q.__experimental_select({
+      "foo > 2": {
+        bar: z.boolean(),
+      },
+    });
+
+    expect(query).toBe("select(foo > 2 => { bar })");
+    expectTypeOf({} as z.infer<typeof schema>).toEqualTypeOf<{
+      bar: boolean;
+    } | null>();
+  });
+
+  it("handles nested selects properly", () => {
+    const nestedSelect = q.__experimental_select({
+      'bar == "thing"': {
+        b: q.string(),
+      },
+      default: {
+        c: z.string(),
+      },
+    });
+
+    const { query, schema } = q.__experimental_select({
+      "foo > 2": nestedSelect,
+      default: {
+        baz: z.string(),
+      },
+    });
+
+    expect(query).toBe(
+      'select(foo > 2 => select(bar == "thing" => { b }, { c }), { baz })'
+    );
+    expectTypeOf({} as z.infer<typeof schema>).toEqualTypeOf<
+      { b: string } | { c: string } | { baz: string }
+    >();
+  });
+});
+
+describe("EntityQuery.select()", () => {
+  describe("with standalone select input", () => {
+    it("updates query and handles nested object types", () => {
+      const standaloneSelect = q.__experimental_select({
+        "foo > 2": {
+          bar: z.boolean(),
+        },
+        default: {
+          baz: z.string(),
+        },
+      });
+      const entityQuery = q("foo").__experimental_select(standaloneSelect);
+
+      expect(entityQuery.query).toBe(
+        "foo{...select(foo > 2 => { bar }, { baz })}"
+      );
+      expectTypeOf({} as z.infer<typeof entityQuery.schema>).toEqualTypeOf<
+        { bar: boolean } | { baz: string }
+      >();
+    });
+
+    it("converts no default condition to empty object", () => {
+      const standaloneSelect = q.__experimental_select({
+        "foo > 2": {
+          bar: z.boolean(),
+        },
+      });
+      const entityQuery = q("foo").__experimental_select(standaloneSelect);
+
+      expect(entityQuery.query).toBe("foo{...select(foo > 2 => { bar })}");
+      expectTypeOf({} as z.infer<typeof entityQuery.schema>).toEqualTypeOf<
+        { bar: boolean } | {}
```

Comment:
I recall that eslint doesn't like the `{}` type, as [it pretty much means `any`](https://www.typescriptlang.org/play?#code/C4TwDgpgBAQgTgQwMYQM5QLxQN4F8DcAsAFAlID2AdqsFAgFzzJqZQCMZVNUARo4inRZsDAORtRuTtVpJ+zIVFGjp3ACbzBrPKtoRNLLJQCuAG1MkSoSFACiAWzCgA8jwBWEJLSwAlT+Tg1AB4aOABLSgBzABooSggANwg4AD4iUmIKGSgAL3oHJxBXDy9WDkyuWhB8xxd3T28cBmNKNQgAMwiINSkK7IAPGsLihtZlXSgAdyG6ksadPu4EmaL60qMzC2IgA). Any way to work around using it?

Reply:
Ended up running into some issues here when testing validation, but updated and wrote some extra tests to cover the fix. We actually did need a `z.record` type in the schema to enforce empty objects in validation:
```ts
const emptyObject = z.record(z.string(), z.never())

// this resolves to record<string, never> like mentioned earlier
type EmptyObject = z.infer<typeof emptyObject>
// ^? Record<string, never>
```